### DISCUSSION TITLE
Conditionals based on Repeater Row Index

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#5654] [General]
I have a Data Provider -> Repeater -> Container situation where i would like to set the Container background to light grey on even rows, and while on odd rows to create the zebra effect, and also i would like the first Row of the Repeater to carry bold fonts while the rest rows would not. 

Is this something currently achievable ?
Is there a way to set / update settings of inside elements based on the enclosing repeater row position ?

**Comment by melohagan** [comment]
HI @poirazis 

This is doable with Conditional UI: https://docs.budibase.com/docs/conditions 

Have a Data Provider with Repeater and two containers:
![Screenshot 2022-04-29 at 16 30 10](https://user-images.githubusercontent.com/101575380/165976242-96539fce-2fe8-4801-98da-38df72ba8579.png)

**Configure conditions** for the light container:
![Screenshot 2022-04-29 at 16 35 05](https://user-images.githubusercontent.com/101575380/165977069-0fc82bc6-b82b-4c04-a005-343c6c4e017e.png)

*JavaScript*
```
return $("Repeater.OddEven.Auto ID") % 2 ? $("Repeater.OddEven.Auto ID") : false
```

Then the dark container will have the exact same condition, but the ternary will be flipped:
```
return $("Repeater.OddEven.Auto ID") % 2 ? false : $("Repeater.OddEven.Auto ID")
```

I'm using *Auto ID* here, but you could use any column that is an incrementing integer.

**Result**
![Screenshot 2022-04-29 at 16 37 21](https://user-images.githubusercontent.com/101575380/165977457-1b2219b5-912d-4806-a46f-48fdbf5c3fba.png)
